target/linux/ramips/image/mt7621.mk

```diff
@@ -142,6 +142,16 @@ define Device/newifi-d1
 endef
 TARGET_DEVICES += newifi-d1
 
+define Device/newifi-d2
+  DTS := Newifi-D2
+  IMAGE_SIZE := $(ralink_default_fw_size_16M)
```

Comment:
ralink_default_fw_size_16M is wrong here. ralink_default_fw_size_16M is 16121856 bytes (0xf60000). But your firmware partition is 0xfb0000 bytes.